apps/recorder/widget.js

```diff
@@ -176,13 +176,16 @@
   };
   let getCSVHeaders = activeRecorders => ["Time"].concat(activeRecorders.map(r=>r.fields));
 
-  let writeLog = function() {
+  let writeLog = function(period, fix) {
     entriesWritten++;
     WIDGETS["recorder"].draw();
     try {
-      var fields = [Math.round(getTime())];
+      console.log("\nfix lat:         " + fix.lat + "\nfix time (ms):   " + fix.time + "\ncpu time (s):    " + getTime());
+      var time = (fix.time&&fix.lat)?fix.time/1000:getTime();
```

Comment:
I'm deliberating on whether to use the GPS fix time for the log entries. 

Here the GPS time is used if the fix also contain coordinates. Otherwise the Bangle system time is used. The effect of this is:
- Before a fix with GPS coordinates is found the system time is used.
- If the GPS is no longer able to determine coordinates, the log will go back to using the system time.

If the difference between GPS time and Bangle system time is large enough this can give the result that a log entry gets a reported time that is older than previous log entries.

I believe an advantage of using GPS fix time is it's reporting becomes independent of the variable time before the fix is acted on by the recorder widget. This should make analytics (speed, etc.) more accurate. But it comes with some kinks as described above, and probably some other consequence I'm not thinking of.

Reply:
I would prefer a monotonic clock for recording, but since the bangle does not have one the system time is the next best thing. Also I if using e.g. [gpsautotime](https://github.com/espruino/BangleApps/tree/master/apps/gpsautotime) both time sources should have the same value.

>  believe an advantage of using GPS fix time is it's reporting becomes independent of the variable time before the fix is acted on by the recorder widget. This should make analytics (speed, etc.) more accurate,

I dcan't follow here. Yes the GPS time is very accurate but I don't think the system time is that inaccurate that it would make a difference.

Also: if in doubt, why not log both times?

Reply:
Yes, I'm not really sure. It's likely you have a better understanding of this.

My current assumption is GPS time is initially based on system time until it starts getting coordinates, at which time I'd thought the time would be given by the satellites. But I have not verified this.

Reply:
System time is not needed since their time is the main thing that the satellites are sending.

Reply:
But until the GPS receiver has contact with satellites it seemed to be using the Bangle system time rounded to nearest second, when I did some testing last night. But again, not really verified/falsified.

Reply:
> GPS time is initially based on system time until it starts getting coordinates

It should report the time the GPS receiver has, which is actually different to the Bangle's time. I think if the battery was totally flat and you restarted you'd find you didn't get any GPS time.

Personally, I think we should just use Bangle.js time - there's really not going to be any difference of more than a few seconds, and the downside of having the GPS time is it could just completely change when you get a fix and that'll throw off everything.

If you think it's something you really need then as @nxdefiant says you could just make a new app with an `app.recorder.js` file which added a new column for GPS time. But we definitely shouldn't add it by default, as it'll just use up even more storage space for recordings.

Reply:
Thanks!

Reply:
I dropped the commit/changes for using GPS time.

Reply:
I dropped the commit/changes for using the GPS time. Thanks for your feedback! :)